src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/Tools/Move-AzureRmVhdVM.ps1

```diff
@@ -0,0 +1,237 @@
+    $query = "Associators of {$Vm} Where AssocClass=Msvm_SystemDevice ResultClass=Msvm_KvpExchangeComponent";
+    $Kvp = gwmi -Namespace root\virtualization\v2 -Query $query -ComputerName $computerNameInput;
+
+    #Write-Host "Guest KVP information for: '$hyperVVMNameInput'";
+
+    # Filter the results
+    try
+    {
+        $osNameItem = $Kvp.GuestIntrinsicExchangeItems | Import-CimXml | where Name -eq "OSName";
+        Write-Output $osNameItem.Data;
+    }
+    catch
+    {
+        Write-Verbose "OS type name not found...";
+    }
+}
+
+function Export-VMWithProgress($computerName, $vmName, $exportDir)
+{
+    # Record the supposed to be exported VHD files first
+    $vmObj = Get-VM -ComputerName $computerName -Name $vmName;
+    if (-not $vmObj.HardDrives -or $vmObj.HardDrives.Count -le 0)
+    {
+        throw "No hard drives can be found from VM '$vmName'...";
+    }
+    [string[]]$vmDiskFileNames = @();
+    foreach ($hd in $vmObj.HardDrives)
+    {
+        $vmDiskFileNames += (Split-Path -Path $hd.Path -Leaf);
+    }
+    
+    # Start the real export process
+    $exportJob = Export-VM -ComputerName $computerName -Name $vmName -Path $exportDir -AsJob;
+    while( $exportJob.State -eq "Running" -or $exportJob.State -eq "NotStarted") 
+    {
+        Write-Host ("[Export] " + $($exportJob.Progress.PercentComplete) + "% completed...");
+        sleep(5);
+    }
+
+    if($exportJob.State -ne "Completed")
+    {
+        Write-Error ("Export-VM did not complete: " + $exportJob.State);
+        throw $exportJob.Error;
+    }
+  
+    # Construct the final VHD file paths
+    [string[]]$vhdFiles = @();
+    foreach ($fileName in $vmDiskFileNames)
+    {
+        $vhdFullName = Join-Path (Join-Path (Join-Path $exportDir $vmName) 'Virtual Hard Disks') $fileName;
+        if (-not (Test-Path $vhdFullName))
+        {
+            throw "The VHD file '$vhdFullName' should have been exported, but it cannot be found.";
+        }
+        $vhdFiles += $vhdFullName;
+    }
+    Write-Host "Exported VHD files: '$vhdFiles'";
+    return $vhdFiles;
+}
+
+function Try-GetHyperVVM($computerName, $hvvmName)
+{
+    if ([string]::IsNullOrEmpty($computerName))
+    {
+        throw "Hyper-V computer name cannot be null or empty: '$computerName'.";
+    }
+    if ([string]::IsNullOrEmpty($hvvmName))
+    {
+        throw "Hyper-V VM name cannot be null or empty: '$hvvmName'.";
+    }
+
+    try
+    {
+        $allVMs = Get-VM -ComputerName $computerName;
+        $hvvm = $allVMs | where {$_.Name -eq $hvvmName};
+        if ($hvvm -ne $null)
+        {
+            Write-Host ("Found Hyper-V VM by name '$hvvmName': " + $hvvm);
+            return $true;
+        }
+        else
+        {
+            Write-Warning ("Cannot find Hyper-V VM by name '$hvvmName'...");
+            return $false;
+        }
+    }
+    catch
+    {
+        throw;
+    }
+}
+
+function Try-LoginAzureRmAccount
+{
+    try
+    {
+        $st = Get-AzureRmVM;
```

Comment:
Check for the Context and throw a meaningful error if it is not available. Do not attempt to log in as part of this cmdlet

Reply:
removed